File: AdminPanel/DataAccess/Entities/Resource.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace StudyBuddySAP.DataAccess.Entities

{
    [BsonIgnoreExtraElements]
    public class Resource
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public ObjectId _id { get; set; }
        public string? title { get; set; }
        public string? description { get; set; }
        public string? link { get; set; }
        public DateTime uploaded_at { get; set; }
        public ObjectId uploaded_by { get; set; }
        public ObjectId group_id { get; set; }
        public int __v { get; set; }

    }

}

File: backend/model/Resource.js
const mongoose = require('mongoose');
const Schema = mongoose.Schema;

const resourceSchema = new Schema({
    title: {
        type: String,
        required: true
    },

    description: {
        type: String
    },

    link: {
        type: String,
        required: true
    },

    uploaded_at: {
        type: Date,
        default: Date.now()
    },

    uploaded_by: {
        type: Schema.Types.ObjectId,
        ref: 'User'
    },

    group_id: {
        type: Schema.Types.ObjectId,
        ref: 'Group'
    }
});

module.exports = mongoose.model('Resource', resourceSchema);